web/src/components/onboarding/OnboardingIndustry.test.tsx

```diff
@@ -40,131 +54,99 @@ describe("<OnboardingIndustry />", () => {
-        renderComponent();
-        selectIndustry("car-service");
-        fireEvent.click(screen.getByText(Config.profileDefaults.STARTING.carService.radioButtonBothText));
-        expect(currentProfileData().carService).toBe("BOTH");
+    const businessPersonas = ["STARTING", "FOREIGN"];
+    let profileData: ProfileData;
+    EssentialQuestions.map((el) => {
+      const validIndustryId = randomFilteredIndustry(el.validationFunc);
+      const nonValidIndustryId = randomNegativeFilteredIndustry(el.validationFunc);
+      profileData = {
+        ...createEmptyProfileData(),
+        industryId: nonValidIndustryId.id,
+      };
+      businessPersonas.map((persona) => {
+        const choices = industrySpecificDataChoices[el.fieldName];
+        const flowConfig = Config.profileDefaults[persona as FlowType];
+        const fieldContent = flowConfig[
+          el.contentFieldName ?? (el.fieldName as keyof typeof flowConfig)
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ] as any;
+
+        it(`displays ${el.contentFieldName ?? el.fieldName} for ${
+          validIndustryId.id
+        } as a ${persona} when selected`, () => {
+          renderComponent(profileData);
+          expect(screen.queryByTestId(`industry-specific-${validIndustryId.id}`)).not.toBeInTheDocument();
+
+          selectIndustry(validIndustryId.id);
+          expect(screen.getByTestId(`industry-specific-${validIndustryId.id}`)).toContainHTML(
+            renderToStaticMarkup(
+              Content({
+                children: fieldContent.description,
+              })
+            )
+          );
+          choices.map((choice) => {
+            fireEvent.click(
+              screen.getByText(
+                fieldContent[
+                  `radioButton${capitalizeFirstLetter(kabobSnakeSentenceToCamelCase(choice.toString()))}Text`
+                ]
+              )
+            );
+            expect(currentProfileData()[el.fieldName]).toEqual(choice);
+          });
+          selectIndustry("generic");
+          expect(screen.queryByTestId(`industry-specific-${validIndustryId.id}`)).not.toBeInTheDocument();
+        });
+
+        it(`sets ${el.contentFieldName ?? el.fieldName} back to ${
+          emptyIndustrySpecificData[el.fieldName]
+        } if they select a different industry`, () => {
+          renderComponent(profileData);
+          selectIndustry(validIndustryId.id);
+          fireEvent.click(
+            screen.getByText(
+              fieldContent[
+                `radioButton${capitalizeFirstLetter(
+                  kabobSnakeSentenceToCamelCase(choices[0].toString())
+                )}Text`
+              ]
+            )
+          );
+          expect(currentProfileData()[el.fieldName]).toEqual(choices[0]);
+          selectIndustry("generic");
+          expect(currentProfileData()[el.fieldName]).toEqual(emptyIndustrySpecificData[el.fieldName]);
+        });
```

Comment:
this is hard to read at points, and maybe we could help that by adding named helper functions like `chooseFirstRadioButton` or `chooseRadioWithContent` but assuming this test works and tests everything the way it appears to, this seems like a huge win